reader_go18_test.go

```diff
@@ -88,6 +88,46 @@ func testGenericReaderRows[Row any](rows []Row) error {
 	return nil
 }
 
+func TestIssue400(t *testing.T) {
+	type B struct {
+		Name string
+	}
+	type A struct {
+		B []B `parquet:",optional"`
+	}
+
+	b := new(bytes.Buffer)
+	w := parquet.NewGenericWriter[A](b)
+	expect := []A{
+		{
+			B: []B{
+				{
+					// 32 bytes random so we can see in the binary parquet if we
+					// actually wrote the value
+					Name: "9e7eb1f0-bbcc-43ec-bfad-a9fac1bb0feb",
+				},
+			},
+		},
+	}
+	_, err := w.Write(expect)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = w.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	r := parquet.NewGenericReader[A](bytes.NewReader(b.Bytes()))
+	values := make([]A, 2)
+	_, err = r.Read(values)
+	if err != nil && err != io.EOF {
```

Comment:
Can someone with more experience with this library verify that this is expected behavior.

Reply:
This seems fine but could we expect a single condition rather than being flexible? (e.g. only test for io.EOF if that's what we expect to see)

Reply:
Alternatively you could also use `parquet.Read[A]` instead of a reader to reduce the about of code?